pkg/R/SQLContext.R

```diff
@@ -1,5 +1,130 @@
 # SQLcontext.R: SQLContext-driven functions
 
+infer_type <- function(x) {
+  if (is.null(x)) {
+    stop("can not infer type from NULL")
+  }
+  type <- switch(class(x),
+                 integer = "long",
```

Comment:
In PySpark and JsonRDD/JsonFile, we changed to use LongType for integer in Python and JSON, because IntegerType is 32bit, but int in Python/JSON is 64bit (in 64-bit machine). It could overflow if we use IntegerType.